crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,125 @@
+use crate::semantic_services::Semantic;
+use biome_analyze::{
+    context::RuleContext, declare_rule, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_js_syntax::{JsBlockStatement, JsLabeledStatement, JsStatementList, AnyJsStatement, JsFunctionDeclaration, JsVariableStatement};
+use biome_js_syntax::JsVariableKind::{Const, Let};
+use biome_rowan::{AstNode, AstNodeList, TextRange};
+
+declare_rule! {
+    /// Disallow unnecessary nested block statements.
+    ///
+    /// > In JavaScript, prior to ES6, standalone code blocks delimited by curly braces do not create a new scope and have no use.
+    /// > In ES6, code blocks may create a new scope if a block-level binding (let and const), a class declaration or a function declaration (in strict mode) are present. A block is not considered redundant in these cases.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
+            return None;
+        }
+
+        if block.statements().is_empty() {
```

Comment:
I think we should not report empty block in `for` statements. It is sometimes use in this way:

```js
function g() {
  let i;
  for(i = 0; f(i); i++) {}
  return i;
}
```

Reply:
Couldn't be true for any loop?

Reply:
I've added this and your other example to the tests

Reply:
> Couldn't be true for any loop?

You are right. Except for `for-of`?

Reply:
At the moment I think this will allow

```js
for (const a of foo) {}
```

Do I need to add a case to catch this?

Reply:
I think it is ok for a first version. We could add in another PR control structures.